### ISSUE TITLE
Ability to rebase a newly-created file

### ISSUE THREAD

**Issue by jonpalmisc** [Vector35/binaryninja-api#5186]
It's not possible to rebase a file created via "New File" since it creates a raw view and those cannot be rebased.

My workflow is as follows:

1. Have some assembly I want to assemble and place at a specific address.
2. Create new file in BN.
3. Use the "Assemble" dialog to assemble the code inside of BN.
4. Create functions, etc. for my code in linear view—the usual stuff.
5. Rebase so it's at the right address.

Attempting to rebase a raw view yields a borderline-too-quiet warning log messaging indicating raw views cannot be rebased.

Ideally there would be a way to promote a raw view to a mapped view easily via the UI so this could be done.

**Comment by plafosse** [comment]
Yeah I think this is annoying the only option we have here is to save the file and re-open them so you get a mapped view.

**Comment by bpotchik** [comment]
Seems like we can add a "New Mapped File" action and create a mapped view right away.

**Comment by bpotchik** [comment]
Adding support for this should resolve this workflow. https://github.com/Vector35/binaryninja-api/issues/4082

**Comment by bpotchik** [comment]
There are a couple of options now:

0. Rebase the new file ('Raw' view). It will automatically create a 'Mapped' view for you and prompt for a base address. Note: It's better to make functions/perform markup in the Mapped view rather than a raw view.

1. Use `New Mapped Data` action. You can assemble/compile directly into the Mapped view and it will write to the underlying view and keep the memory map in sync.

2. You can use the 'Add Memory Region' dialog in a Mapped View to create a region with a fixed size. You can assemble/compile to it.

Fixed in `5.1.7927-dev`.
